app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt

```diff
@@ -26,26 +27,33 @@ import androidx.compose.ui.unit.dp
 import coil.compose.AsyncImage
 import com.github.se.cyrcle.R
 import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.review.ReviewViewModel
 import com.github.se.cyrcle.model.user.UserViewModel
 import com.github.se.cyrcle.ui.navigation.NavigationActions
 import com.github.se.cyrcle.ui.navigation.Screen
 import com.github.se.cyrcle.ui.theme.ColorLevel
 import com.github.se.cyrcle.ui.theme.atoms.Button
+import com.github.se.cyrcle.ui.theme.atoms.ScoreStars
 import com.github.se.cyrcle.ui.theme.atoms.Text
 import com.github.se.cyrcle.ui.theme.bold
 import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
 
+@SuppressLint("StateFlowValueCalledInComposition")
 @Composable
-fun CardScreen(
+fun ParkingDetailsScreen(
     navigationActions: NavigationActions,
     parkingViewModel: ParkingViewModel,
-    userViewModel: UserViewModel
+    userViewModel: UserViewModel,
+    reviewViewModel: ReviewViewModel
 ) {
   val selectedParking =
       parkingViewModel.selectedParking.collectAsState().value
           ?: return Text(stringResource(R.string.no_selected_parking_error))
-
   val userSignedIn = userViewModel.isSignedIn.collectAsState(false)
+  reviewViewModel.getReviewsByParking(selectedParking.uid)
+  selectedParking.nbReviews = reviewViewModel.parkingReviews.value.size
```

Comment:
My hand was kind of forced in this respect by the Map. A PR from last week broke the expected functionality, and so the only solution was to re-update from the database every time the screen is called. You are correct that this is not an ideal fix and using the reviewViewModel isn't ideal, but it's the only one I can feasibly think of right now to fix this.

Reply:
After reflection, I'll reduce the scope of the PR and revert the screen to what it was before. I'll work on a fix at a later date

Reply:
Hey just dropping by to say it's better if you let the bug in your PR and make an issue so that we can fix it later, rather than trying to do some patchwork. This will cost us more in the long term.

Btw I think I know exactly why the nbReviews gets out of sync, it's my bad not Julien. 
I forgot to update the cached data when you call .updateReviewScore()  in the viewmodel

Reply:
Please read my above comment, we are in agreement on this